omero/autogen_db_version.py

```diff
@@ -5,23 +5,25 @@
 import os
 import re
 import sys
-
+from path import path
+from omero.cli import CLI
+from omero.install.config_parser import PropertyParser
 
 def get_mmp(sqlfile):
     m = re.search('.*/?OMERO(\d+)\.(\d+)(\w*)__(\d+)', sqlfile)
     mmp = (int(m.group(1)), int(m.group(2)), m.group(3), int(m.group(4)))
     return mmp
 
 
-serverdir = sys.argv[1]
+serverdir = path(sys.argv[1])
+
 required = {'omero.db.version': None, 'omero.db.patch': None,
             'versions.bioformats': None}
-with open(os.path.join(serverdir, 'etc', 'omero.properties')) as f:
-    for line in f.readlines():
-        m = re.match('([\w\.]+)=(.*)', line.strip())
-        if m and m.group(1) in required:
-            required[m.group(1)] = m.group(2)
-            print '%s = "%s"' % (m.group(1).replace('.', '_'), m.group(2))
+cli = CLI()
+property_lines = cli.get_config_property_lines(serverdir)
+for property in PropertyParser().parse_lines(property_lines):
+    if property.key in required:
+        required[property.key] = property.val
```

Comment:
(Pity you can't just use `PropertyParser`'s `data()` getter but it chokes on `product.name=OMERO.server`.)

Reply:
Do we know _why_ it chokes?

Reply:
https://github.com/openmicroscopy/openmicroscopy/blob/v5.5.0-m5/components/tools/OmeroPy/src/omero/install/config_parser.py#L320

Reply:
Gotcha. Think that property can now go away ;)

Reply:
the property was needed for one of the jobs. this will need to be reviewed before we remove it.